### PR TITLE
Fix CVE-2023-50711 in cloud-hypervisor

### PR DESCRIPTION
**Pull request by sindhu-karri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Cherry-pick this [PR](https://github.com/microsoft/CBL-Mariner/pull/7269) from main to fast-track branch
PR patches the vendor packages vmm-sys-util, vhost and versionize to fix [CVE-2023-50711](https://nvd.nist.gov/vuln/detail/CVE-2023-50711)

###### Change Log  

- Patch [CVE-2023-50711](https://nvd.nist.gov/vuln/detail/CVE-2023-50711) in vendor/vmm-sys-util, vendor/vhost, vendor/versionize

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-50711

###### Test Methodology

- Pipeline build id: xxxx
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=487210&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=488218&view=results

### CODE REVIEW

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 1]
```diff
@@ -75,6 +80,12 @@ tar xf %{SOURCE1}
 pushd vendor/libz-sys/src/zlib
 %patch0 -p1
 popd
+pushd vendor/vmm-sys-util
+%patch1 -p1
+popd
+%patch2 -p1	
+%patch3 -p1
+%patch4 -p1
```
Comment (PawelWMS) [inline SPECS/cloud-hypervisor/cloud-hypervisor.spec:88]: Could we just bundle this into one patch and also update the paths in `Patch1`, so we can drop the pushd-pop stuff?

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 2]
```diff
@@ -19,9 +19,14 @@ Source0:        https://github.com/cloud-hypervisor/cloud-hypervisor/archive/ref
...
 #   cargo vendor > config.toml
 #   tar -czf %{name}-%{version}-cargo.tar.gz vendor/
+# rename the tarball to %{name}-%{version}-cargo.tar.gz when updating version
 Source1:        %{name}-%{version}-cargo.tar.gz
 Source2:        config.toml
 Patch0:         CVE-2023-45853.patch
+Patch1:         CVE-2023-50711-vmm-sys-util-fix1.patch 
+Patch2:         CVE-2023-50711-vmm-sys-util-fix2-cargo.patch
+Patch3:         CVE-2023-50711-vhost.patch
+Patch4:         CVE-2023-50711-versionize.patch
```
Comment (PawelWMS) [inline SPECS/cloud-hypervisor/cloud-hypervisor.spec:29]: @nisamson, will a patch file `CVE-XXXX-YYYY-something.patch` be also interpreted correctly by our CVE scanning logic?

**Comment by PawelWMS** [comment]
The one failing PR check is a false-positive from the Cred-Scan. The issue is being fixed in #7298 (will need porting to fast-track branches) but for now an admin can override the check and merge then changes, when we're through the comments.

**Review by jslobodzian — APPROVED** [review]

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec (hunk 2 above)
Comment (sindhu-karri) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec]: Yes, the patch is interpreted correctly.

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec (hunk 1 above)
Comment (PawelWMS) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec]: Got it. Please just pick the fast-track version while you resolve the merge conflicts in the cherry-pick PR once this is one is merged.

**Comment by PawelWMS** [comment]
Checked the package build and test run - the only failing step is the cred-scan part mentioned in the earlier comments. There are no actual password there, just something used in test and the test files containing these "secrets" are already removed in `main`. Merging.

**Pull request merged**